e2e-tests/demand-backup/run

```diff
@@ -6,6 +6,23 @@ set -o xtrace
 test_dir=$(realpath $(dirname $0))
 . ${test_dir}/../functions
 
+run_recovery_from_source() {
+    desc "aws restore with backup source"
+    stext=$(kubectl get PerconaXtraDBClusterBackup on-demand-backup-aws-s3 -o json | jq -c '.status')
+    content=$(jq --argjson s $stext '.spec.backupSource=$s' $test_dir/conf/restore-from-source-backup-aws-s3.json)
+    cat > $test_dir/conf/restore-from-source-backup-aws-s3.json <<<$content
+    kubectl_bin apply -f $test_dir/conf/restore-from-source-backup-aws-s3.json
```

Comment:
keep it simple
```suggestion
    stext=$(kubectl_bin get pxc-backup on-demand-backup-aws-s3 -o json | jq -c '.status')
    jq --argjson s "$stext" '.spec.backupSource=$s' $test_dir/conf/restore-from-source-backup-aws-s3.json \
        | kubectl apply -f -
    
```

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/556/commits/edfe5858f06e192094e7703e3ba62b0afea5647e